File: python/02.py
filename = "02.txt"

all_package_sizes = 0
ribbon_length = 0


def find_paper_size(length, width, height):
    l_area = (length * width)
    w_area = (width * height)
    h_area = (length * height)

    shortest = min(l_area, w_area, h_area)

    if shortest == l_area:
        rl = (2 * length) + (2 * width)
    elif shortest == w_area:
        rl = (2 * width) + (2 * height)
    else:
        rl = (2 * length) + (2 * height)

    rl = rl + (length * width * height)

    area = (2 * l_area) + (2 * w_area) + (2 * h_area) + min(l_area, w_area, h_area)

    return (rl, area)

with open(filename, "r") as f:
    for l in f.readlines():
        (l, w, h) = l.split("x")
        (length, area) = find_paper_size(int(l), int(w), int(h))
        ribbon_length = ribbon_length + length
        all_package_sizes = all_package_sizes + area

print "we need", all_package_sizes, "sqft of paper"
print "we need", ribbon_length, "feet of ribbon"

File: javascript/02.js
var fs = require('fs');

var filename = '02.txt';
var contents = fs.readFileSync(filename).toString().trim().split('\n');
var paper_area = 0;
var ribbon_length = 0;

console.log(contents.length);

for (var i = 0; contents.length; i++) {
  if (!contents[i]) {
    break;
  }

  var dimensions = contents[i];
  var box = dimensions.split('x');
  var l = parseInt(box[0]);
  var w = parseInt(box[1]);
  var h = parseInt(box[2]);
  var lw = l * w;
  var wh = w * h;
  var lh = l * h;

  var shortest = Math.min(lw, wh, lh);

  if (shortest === lw) {
    ribbon_length += (2 * l) + (2 * w);
  } else if (shortest === wh) {
    ribbon_length += (2 * w) + (2 * h);
  } else if (shortest === lh) {
    ribbon_length += (2 * l) + (2 * h);
  }

  ribbon_length += (l * w * h);

  paper_area += (2 * lw) + (2 * wh) + (2 * lh) + Math.min(lw, wh, lh);
}

console.log("paper area required is " + paper_area);
console.log("ribbon length is " + ribbon_length);
